azure-iot-device/tests/iothub/test_sync_clients.py

```diff
@@ -1503,6 +1526,144 @@ class TestIoTHubDeviceClientReceiveTwinDesiredPropertiesPatch(
     pass
 
 
+@pytest.mark.describe("IoTHubDeviceClient (Synchronous) -.get_storage_info()")
+class TestIoTHubDeviceClientGetStorageInfo(WaitsForEventCompletion, IoTHubDeviceClientTestsConfig):
+    @pytest.mark.it("Begins a 'get_storage_info' HTTPPipeline operation")
+    def test_calls_pipeline_get_storage_info(self, client, http_pipeline):
+        fake_blob_name = "__fake_blob_name__"
+        client.get_storage_info(fake_blob_name)
+        assert http_pipeline.get_storage_info.call_count == 1
+        assert http_pipeline.get_storage_info.call_args[0][0] is fake_blob_name
+
+    @pytest.mark.it(
+        "Waits for the completion of the 'get_storage_info' pipeline operation before returning"
+    )
+    def test_waits_for_pipeline_op_completion(
+        self, mocker, client_manual_cb, http_pipeline_manual_cb
+    ):
+        fake_blob_name = "__fake_blob_name__"
+
+        self.add_event_completion_checks(
+            mocker=mocker,
+            pipeline_function=http_pipeline_manual_cb.get_storage_info,
+            kwargs={"storage_info": "__fake_storage_info__"},
+        )
+
+        client_manual_cb.get_storage_info(fake_blob_name)
+
+    @pytest.mark.it(
+        "Raises a client error if the `get_storage_info` pipeline operation calls back with a pipeline error"
+    )
+    @pytest.mark.parametrize(
+        "pipeline_error,client_error",
+        [
+            pytest.param(
+                pipeline_exceptions.ProtocolClientError,
+                client_exceptions.ClientError,
+                id="ProtocolClientError->ClientError",
+            ),
```

Comment:
Is this really the only expected error?

Reply:
as far as i can see yes, only this or an unhandled error

---
In reply to: [358449708](https://github.com/Azure/azure-iot-sdk-python/pull/403#discussion_r358449708) [](ancestors = 358449708)